docs/Routing.md

```diff
@@ -209,4 +198,43 @@ export const StoreAdmin = () => (
 );
 ```
 
- This will let react-admin build absolute URLs including the sub path.
\ No newline at end of file
+This will let react-admin build absolute URLs including the sub path.
+
+## Troubleshooting
+
+When using custom routing configurations, you may encounter strange error messages like:
+
+> `useLocation()` may be used only in the context of a `<Router>` component
+
+or
+
+> `useNavigate()` may be used only in the context of a `<Router>` component
+
+or 
+
+> `useRoutes()` may be used only in the context of a `<Router>` component
+
+or 
+
+> `useHref()` may be used only in the context of a `<Router>` component.
+
+or
+
+> `<Route>` may be used only in the context of a `<Router>` component
+
+These errors can happen if you added `react-router` and/or `react-router-dom` to your dependencies, and didn't use the same version as react-admin. In that case, your application has two versions of react-router, and the calls you add can't see the react-admin routing context.
+
+To fix this error, make sure to use the same version as react-admin, and deduplicate them using yarn's `resolutions` or npm's `overrides`.
```

Comment:
Would it be worth it to mention that we can use `npm list react-router` or `npm list react-router-dom` to both see the version used by react-admin and also check for a duplicate version?